chain/jsonrpc/src/lib.rs

```diff
@@ -621,13 +628,11 @@ impl JsonRpcHandler {
             .map_err(|err| err.into())))
     }
 
-    async fn block(&self, params: Option<Value>) -> Result<Value, RpcError> {
-        let block_reference = if let Ok((block_id,)) = parse_params::<(BlockId,)>(params.clone()) {
-            BlockReference::BlockId(block_id)
-        } else {
-            parse_params::<BlockReference>(params)?
-        };
-        jsonify(self.view_client_addr.send(GetBlock(block_reference)).await)
+    async fn block(
```

Comment:
why is this method refactored but not other methods?

Reply:
@bowenwang1996 It is to minimize the scope. I asked @khorolets to start with a single method to justify the refactoring (extracting parts into separate *-primitives crates), and then we can purely focus on jsonrpc refactoring scaling the work of refactoring the methods and also paving the road for getting rid of `Unreachable` error states (see #3851)